soccer/MainWindow.cpp

```diff
@@ -718,14 +703,50 @@ void MainWindow::updateStatus() {
     RJ::Time curTime = RJ::now();
 
     // Determine if we are receiving packets from an external referee
-    bool haveExternalReferee =
-        (curTime - ps.lastRefereeTime) < RJ::Seconds(0.5);
+    bool haveExternalReferee = (curTime - ps.lastRefereeTime) < RJ::Seconds(1);
+
+    std::vector<int> validIds = _processor->state()->ourValidIds();
+
+    for (int i = 1; i <= Num_Shells; i++) {
+        if (std::find(validIds.begin(), validIds.end(), i - 1) !=
+            validIds.end()) {
+            // The list starts with None so i is 1 higher than the shell id
+            _ui.goalieID->setItemData(i, true, Qt::UserRole);
```

Comment:
This line seems to be broken.

When I bring a robot off the field, and back onto the field, the dropdown stays disabled. 

The logic is correct (it's running this line at the correct times), but this line does not actually re-enable that dropdown. I'm looking at it now.